packages/addon/src/channel_selector.ts

```diff
@@ -35,40 +40,95 @@ function fillChannels(data: ChannelDetails[], selected: string | null, messageCl
 
 window.addEventListener("load", () => {
   const parent = window.parent;
-  const logo = document.getElementById("logo")!!
+  const logo = document.getElementById("logo")!;
 
   const mc = new MessageChannel();
-  const myPort = mc.port1
-  myPort.start()
-
-  parent.postMessage({ type: "SelectorMessageInitialize" }, "*", [mc.port2]);
-
-  function changeSize(expanded: boolean) {
-    document.body.setAttribute("data-expanded", "" + expanded);
-    myPort.postMessage({ type: "SelectorMessageResize", expanded })
-  }
-
-  myPort.addEventListener("message", (e: MessageEvent) => {
-    if (e.data.type == 'SelectorMessageChannels') {
-      const details = e.data as SelectorMessageChannels
-      // console.log(JSON.stringify("CHANNEL DETAILS: " + JSON.stringify(details)))
-      channels = details.channels
-      channelId = details.selected
-
-      const selectedColor = (channelId ? (channels.find(c => c.id == channelId)?.displayMetadata?.color) : null) ?? 'black'
-      logo.style.fill = selectedColor
+  const myPort = mc.port1;
+  myPort.start();
+  myPort.onmessage = ({data}) => {
+    console.debug("Received message: ", data);
+    switch(data.type){
+      case "iframeHandshake": {
+        collapse();
+        break;
+      }
+      case "iframeChannels": {
```

Comment:
Needs updating to the `fdc3UserInterface*` messsages after we changed the name(s)

```suggestion
      case "fdc3UserInterfaceChannels": {
```

Reply:
@kriswest those changes aren't merged into `fdc3-for-web-impl` so @julianna-ciq's code is correct as it stands.